### ISSUE TITLE
baseURL that has a few path levels get truncated

### ISSUE THREAD

**Issue by nnhubbard** [AFNetworking/AFOAuth1Client#42]
When using a `baseURL` such as `https://staging.geocaching.com/oAuth/mobileoauth.ashx/` it gets cut down to `https://staging.geocaching.com/` which makes requests not work since `https://staging.geocaching.com/request_token` is not a valid page. 

Why is this happening?

**Comment by nnhubbard** [comment]
Looks to me like the issue is in AFHTTPClient.m around line 452: 

`NSURL *url = [NSURL URLWithString:path relativeToURL:self.baseURL];`

Shouldn't it be:

`NSURL *url = [self.baseURL URLByAppendingPathComponent:path];`

**Comment by mattt** [comment]
As per the [documentation of AFHTTPClient](http://cocoadocs.org/docsets/AFNetworking/1.2.1/Classes/AFHTTPClient.html):

> ## URL Construction Using Relative Paths
> 
> Both `-requestWithMethod:path:parameters:` and `-multipartFormRequestWithMethod:path:parameters:constructingBodyWithBlock:` construct URLs from the path relative to the `-baseURL`, using `NSURL +URLWithString:relativeToURL:`. Below are a few examples of how `baseURL` and relative paths interact:

``` objective-c
    NSURL *baseURL = [NSURL URLWithString:@"http://example.com/v1/"];  
    [NSURL URLWithString:@"foo" relativeToURL:baseURL];                  // http://example.com/v1/foo
    [NSURL URLWithString:@"foo?bar=baz" relativeToURL:baseURL];          // http://example.com/v1/foo?bar=baz
    [NSURL URLWithString:@"/foo" relativeToURL:baseURL];                 // http://example.com/foo
    [NSURL URLWithString:@"foo/" relativeToURL:baseURL];                 // http://example.com/v1/foo
    [NSURL URLWithString:@"/foo/" relativeToURL:baseURL];                // http://example.com/foo/
    [NSURL URLWithString:@"http://example2.com/" relativeToURL:baseURL]; // http://example2.com/
```

**Comment by mattt** [comment]
tl;dr Drop the leading slash, or else your baseURL path will get stripped.
